pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -10004,3 +10004,61 @@ spec:
 		})
 	}
 }
+
+func TestReconcile_CancelUnscheduled(t *testing.T) {
+	pipelineRunName := "cancel-test-run"
+	prs := []*v1beta1.PipelineRun{parse.MustParsePipelineRun(t, `metadata:
+  name: cancel-test-run
+  namespace: foo
+spec:
+  pipelineSpec:
+    tasks:
+      - name: wait-1
+        taskSpec:
+          apiVersion: example.dev/v0
+          kind: Wait
+          params:
+            - name: duration
+              value: 1h
+      - name: wait-2
+        runAfter:
+          - wait-1
+        taskSpec:
+          apiVersion: example.dev/v0
+          kind: Wait
+          params:
+            - name: duration
+              value: 10s
+      - name: wait-3
+        runAfter:
+          - wait-1
+        taskRef:
+          name: hello-world
+`)}
+
+	ts := []*v1beta1.Task{simpleHelloWorldTask}
+
+	cms := []*corev1.ConfigMap{withCustomTasks(newFeatureFlagsConfigMap())}
```

Comment:
Would it make sense to make this a matrix test and check also the case

```golang
cms := []*corev1.ConfigMap{withCustomTasks(withEmbeddedStatus(newFeatureFlagsConfigMap(), config.MinimalEmbeddedStatus))}
```

Reply:
Good call!

Reply:
And done.